### PR TITLE
docs: add audit logging docs to agent configuration

### PR DESCRIPTION
**Pull request by mikemorris**
Pulled from original source in https://github.com/hashicorp/consul-enterprise/pull/360/files#diff-004448710778abf0ba756b61b5e1736a and removed the TODOs.

Not sure if there's any style/conventions guidance for the example code block, but the HCL appears to be rendered/highlighted as expected.

@jescalan It appears that nested auto-generated hash links aren't prefixed with parents hashes as they would have been structured previously, e.g. the `audit` > `sink` > `type` nested option link is created as `#sink` instead of `#audit_sink_type`, which may be more prone to collisions.

### CODE REVIEW

File: website/pages/docs/agent/options.mdx [hunk 1]
```diff
@@ -808,6 +808,53 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
...
+          rotate_duration = "24h"
+          rotate_max_files = 15
+          rotate_bytes = 25165824
+        }
+    }
+    ```
+
+    The following sub-keys are available:
+
+    - `enable` - Controls whether Consul logs out each time a user
```
Comment (mkeeler) [inline website/pages/docs/agent/options.mdx:831]:
```suggestion
    - `enabled` - Controls whether Consul logs out each time a user
```

File: website/pages/docs/agent/options.mdx [hunk 2]
```diff
@@ -808,6 +808,53 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
 
 - `advertise_addr_wan` Equivalent to the [`-advertise-wan` command-line flag](#_advertise-wan).
 
+- `audit` <EnterpriseAlert inline /> - Added in Consul 1.8, the audit object allow users to enable auditing
+    and configure a sink and filters for their audit logs.
+
+    ```
```
Comment (jescalan) [inline website/pages/docs/agent/options.mdx:814]: This code block should have a language tag

File: website/pages/docs/agent/options.mdx [hunk 3]
```diff
@@ -808,6 +808,53 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
...
+    }
+    ```
+
+    The following sub-keys are available:
+
+    - `enable` - Controls whether Consul logs out each time a user
+    performs an operation. ACLs must be enabled to use this feature. Defaults to `false`.
+
+    - `sink` - This object provides users configuration for where Consul
+    logs auditing events to. Sink is an object containing keys to sink objects, where the key is the name of the sink.
```
Comment (preetapan) [inline website/pages/docs/agent/options.mdx:835]: should this be `for where Consul auditing events go`?

File: website/pages/docs/agent/options.mdx [hunk 4]
```diff
@@ -808,6 +808,53 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
...
+        The following keys are valid:
+        - `best-effort` - Consul only supports `best-effort` event delivery.
+      - `rotate_duration` - Specifies the
+        interval by which the system rotates to a new log file. At least one of `rotate_duration` or `rotate_bytes`
+        must be configured to enable audit logging.
+      - `rotate_max_files` - Defines the
+        limit that Consul should follow before it deletes old log files.
+      - `rotate_bytes` - Specifies how large an
+        individual log file can grow before Consul rotates to a new file. At least one of `rotate_bytes` or
+        `rotate_duration` or  must be configured to enable audit logging.
```
Comment (preetapan) [inline website/pages/docs/agent/options.mdx:856]: extra `or` after `rotate_duration`

File: website/pages/docs/agent/options.mdx (hunk 2 above)
Comment (mikemorris) [inline website/pages/docs/agent/options.mdx:814]:
```suggestion
    ```hcl
```

Comment (mikemorris) [reply website/pages/docs/agent/options.mdx:814]: @jescalan like this? is HCL a supported language?

File: website/pages/docs/agent/options.mdx (hunk 4 above)
Comment (mikemorris) [inline website/pages/docs/agent/options.mdx:856]:
```suggestion
        `rotate_duration` must be configured to enable audit logging.
```

File: website/pages/docs/agent/options.mdx (hunk 3 above)
Comment (mikemorris) [reply website/pages/docs/agent/options.mdx]: Reworded to remove dangling participle.

**Pull request merged**